time-series/time-series-api/src/main/java/com/powsybl/timeseries/ast/NodeCalcEvaluator.java

```diff
@@ -93,6 +103,23 @@ public NodeCalc iterate(MaxNodeCalc nodeCalc, DoubleMultiPoint multiPoint) {
         return nodeCalc.getChild();
     }
 
+    @Override
+    public Double visit(CachedNodeCalc nodeCalc, DoubleMultiPoint arg, Double child) {
+        double childValue;
+        if (child == null) {
```

Comment:
Looking at the way it is implemented, do you really need to add CachedNodeCalc to the tree ? As far as you have the set of NodeCalc that are duplicated you can directly manage the caching in for instance the binary op node ?

Reply:
I think it might be useful to keep it like this since it allows the users to do other things specifically on those nodes, without having to keep the set of duplicated nodes in memory. For example, if a user wants to change how the nodes are printed, or if s.he adds a new visitor to export the tree and wants to avoid duplication in the export, etc.

Reply:
Or else, we can indeed choose to not use this specific type of nodecalc and force the users to compute again the set of duplicated nodes if they need it somewhere

Reply:
ok